test/typescript.test.ts

```diff
@@ -391,3 +391,22 @@ test.serial('validation errors', async (t) => {
     }
   }
 });
+
+test.serial(
+  'return type',
+  async (t) => {
+    const testInstance = await nohm.factory<UserMockup>('UserMockup');
+    const diffBeforeUpdate = testInstance.propertyDiff('name');
+
+    // update the property
+    testInstance.property('name','testName');
+    const diffAfterUpdate = testInstance.propertyDiff('name');
+
+    t.deepEqual(diffBeforeUpdate, [], 'return without undefined');
+    t.deepEqual(diffAfterUpdate, [{
+      after: 'testName',
+      before: 'defaultName',
+      key: 'name',
+    }], 'return with difference in property');
+  },
+);
```

Comment:
Hmmm, I think this test would not have failed previously, right? The only real change is the return type of an item inside the array. So you'd have to make a test that loops over the diff and explicitly tries to access an item in it.

Something like 

```typescript
diffBeforeUpdate.forEach((diffItem) => { 
  // should never be called, but also shouldn't cause a TS error
  console.log('diff had item', diffItem);
  t.fail('should never reach this');
});
```

Reply:
interesting, the resultant from the earlier version was returning `[undefined]` (as the JS func which doesn't return anything is undefined for return type), and hence this test would have failed for deepEquals when compared with [] on the previous version.